--- binutils-2.41/ld/emulparams/elf_x86_64.sh
+++ binutils-2.41-patched/ld/emulparams/elf_x86_64.sh
@@ -43,13 +43,6 @@
 # Linux/Solaris modify the default library search path to first include
 # a 64-bit specific directory.
 case "$target" in
-  x86_64*-linux*|i[3-7]86-*-linux-*)
-    case "$EMULATION_NAME" in
-      *64*)
-	LIBPATH_SUFFIX=64
-	;;
-    esac
-    ;;
   *-*-solaris2*)
     LIBPATH_SUFFIX=/amd64
     ELF_INTERPRETER_NAME=\"/lib/amd64/ld.so.1\"
