--- a/e2fsprogs/fsck.c
+++ b/e2fsprogs/fsck.c
@@ -60,8 +60,13 @@
 //usage:     "\n	-t TYPE	List of filesystem types to check"
 
 #include "libbb.h"
-#include "volume_id.h"
 #include "common_bufsiz.h"
+
+#if ENABLE_FEATURE_MOUNT_LABEL
+# include "volume_id.h"
+#else
+# define resolve_mount_spec(fsname) ((void)0)
+#endif
 
 /* "progress indicator" code is somewhat buggy and ext[23] specific.
  * We should be filesystem agnostic. IOW: there should be a well-defined
