--- glibc-2.38/sysdeps/mach/hurd/shlib-versions
+++ glibc-2.38-patched/sysdeps/mach/hurd/shlib-versions
@@ -1,4 +1,4 @@
-DEFAULT			GLIBC_2.2.6
+DEFAULT			GLIBC_2.38
 
 libm=6
 
--- glibc-2.38/sysdeps/unix/sysv/linux/aarch64/shlib-versions
+++ glibc-2.38-patched/sysdeps/unix/sysv/linux/aarch64/shlib-versions
@@ -1,4 +1,4 @@
-DEFAULT			GLIBC_2.17
+DEFAULT			GLIBC_2.38
 
 %ifdef HAVE_AARCH64_BE
 ld=ld-linux-aarch64_be.so.1
--- glibc-2.38/sysdeps/unix/sysv/linux/arc/shlib-versions
+++ glibc-2.38-patched/sysdeps/unix/sysv/linux/arc/shlib-versions
@@ -1,4 +1,4 @@
-DEFAULT                 GLIBC_2.32
+DEFAULT                 GLIBC_2.38
 
 %ifdef HAVE_ARC_BE
 ld=ld-linux-arceb.so.2
--- glibc-2.38/sysdeps/unix/sysv/linux/arm/shlib-versions
+++ glibc-2.38-patched/sysdeps/unix/sysv/linux/arm/shlib-versions
@@ -1,4 +1,4 @@
-DEFAULT			GLIBC_2.4
+DEFAULT			GLIBC_2.38
 
 %ifdef HAVE_ARM_PCS_VFP
 # The EABI-derived hard-float ABI uses a new dynamic linker.
--- glibc-2.38/sysdeps/unix/sysv/linux/csky/shlib-versions
+++ glibc-2.38-patched/sysdeps/unix/sysv/linux/csky/shlib-versions
@@ -1,4 +1,4 @@
-DEFAULT			GLIBC_2.29
+DEFAULT			GLIBC_2.38
 
 %if CSKYABI == 2 && CSKY_HARD_FLOAT_ABI == 1
 ld=ld-linux-cskyv2-hf.so.1
--- glibc-2.38/sysdeps/unix/sysv/linux/hppa/shlib-versions
+++ glibc-2.38-patched/sysdeps/unix/sysv/linux/hppa/shlib-versions
@@ -1,11 +1,11 @@
-libm=6			GLIBC_2.2
+libm=6			GLIBC_2.38
 
-libc=6			GLIBC_2.2
+libc=6			GLIBC_2.38
 
-ld=ld.so.1		GLIBC_2.2
+ld=ld.so.1		GLIBC_2.38
 
-libBrokenLocale=1	GLIBC_2.2
+libBrokenLocale=1	GLIBC_2.38
 
-libpthread=0		GLIBC_2.2
+libpthread=0		GLIBC_2.38
 
-libc_malloc_debug=0	GLIBC_2.2
+libc_malloc_debug=0	GLIBC_2.38
--- glibc-2.38/sysdeps/unix/sysv/linux/ia64/shlib-versions
+++ glibc-2.38-patched/sysdeps/unix/sysv/linux/ia64/shlib-versions
@@ -1,10 +1,10 @@
-libc=6.1		GLIBC_2.2
-libm=6.1		GLIBC_2.2
+libc=6.1		GLIBC_2.38
+libm=6.1		GLIBC_2.38
 
-ld=ld-linux-ia64.so.2	GLIBC_2.2
+ld=ld-linux-ia64.so.2	GLIBC_2.38
 
-libBrokenLocale=1	GLIBC_2.2
+libBrokenLocale=1	GLIBC_2.38
 
-libpthread=0		GLIBC_2.2
+libpthread=0		GLIBC_2.38
 
-libc_malloc_debug=0	GLIBC_2.2
+libc_malloc_debug=0	GLIBC_2.38
--- glibc-2.38/sysdeps/unix/sysv/linux/loongarch/shlib-versions
+++ glibc-2.38-patched/sysdeps/unix/sysv/linux/loongarch/shlib-versions
@@ -1,4 +1,4 @@
-DEFAULT		GLIBC_2.36
+DEFAULT		GLIBC_2.38
 
 %if LOONGARCH_ABI_GRLEN == 64 && LOONGARCH_ABI_FRLEN == 64
 ld=ld-linux-loongarch-lp64d.so.1
--- glibc-2.38/sysdeps/unix/sysv/linux/m68k/coldfire/shlib-versions
+++ glibc-2.38-patched/sysdeps/unix/sysv/linux/m68k/coldfire/shlib-versions
@@ -1 +1 @@
-DEFAULT                 GLIBC_2.4
+DEFAULT                 GLIBC_2.38
--- glibc-2.38/sysdeps/unix/sysv/linux/microblaze/shlib-versions
+++ glibc-2.38-patched/sysdeps/unix/sysv/linux/microblaze/shlib-versions
@@ -1 +1 @@
-DEFAULT			GLIBC_2.18
+DEFAULT			GLIBC_2.38
--- glibc-2.38/sysdeps/unix/sysv/linux/mips/shlib-versions
+++ glibc-2.38-patched/sysdeps/unix/sysv/linux/mips/shlib-versions
@@ -1,27 +1,27 @@
-libm=6			GLIBC_2.0 GLIBC_2.2
+libm=6			GLIBC_2.38
 
 # Working mips versions were never released between 2.0 and 2.2.
-libc=6			GLIBC_2.0 GLIBC_2.2
+libc=6			GLIBC_2.38
 
 %ifdef HAVE_MIPS_NAN2008
-ld=ld-linux-mipsn8.so.1	GLIBC_2.0 GLIBC_2.2
+ld=ld-linux-mipsn8.so.1	GLIBC_2.38
 %else
-ld=ld.so.1		GLIBC_2.0 GLIBC_2.2
+ld=ld.so.1		GLIBC_2.38
 %endif
-libdl=2			GLIBC_2.0 GLIBC_2.2
+libdl=2			GLIBC_2.38
 
-libresolv=2		GLIBC_2.0 GLIBC_2.2
+libresolv=2		GLIBC_2.38
 
-libnss_files=2		GLIBC_2.0 GLIBC_2.2
-libnss_dns=2		GLIBC_2.0 GLIBC_2.2
-libnss_compat=2		GLIBC_2.0 GLIBC_2.2
-libnss_ldap=2		GLIBC_2.0 GLIBC_2.2
-libnss_hesiod=2		GLIBC_2.0 GLIBC_2.2
+libnss_files=2		GLIBC_2.38
+libnss_dns=2		GLIBC_2.38
+libnss_compat=2		GLIBC_2.38
+libnss_ldap=2		GLIBC_2.38
+libnss_hesiod=2		GLIBC_2.38
 
-libnsl=1		GLIBC_2.0 GLIBC_2.2
+libnsl=1		GLIBC_2.38
 
-librt=1			GLIBC_2.0 GLIBC_2.2
+librt=1			GLIBC_2.38
 
-libpthread=0		GLIBC_2.0 GLIBC_2.2
+libpthread=0		GLIBC_2.38
 
-libc_malloc_debug=0	GLIBC_2.0 GLIBC_2.2
+libc_malloc_debug=0	GLIBC_2.38
--- glibc-2.38/sysdeps/unix/sysv/linux/nios2/shlib-versions
+++ glibc-2.38-patched/sysdeps/unix/sysv/linux/nios2/shlib-versions
@@ -1,2 +1,2 @@
-DEFAULT                 GLIBC_2.21
+DEFAULT                 GLIBC_2.38
 ld=ld-linux-nios2.so.1
--- glibc-2.38/sysdeps/unix/sysv/linux/or1k/shlib-versions
+++ glibc-2.38-patched/sysdeps/unix/sysv/linux/or1k/shlib-versions
@@ -1,2 +1,2 @@
-DEFAULT                 GLIBC_2.35
+DEFAULT                 GLIBC_2.38
 ld=ld-linux-or1k.so.1
--- glibc-2.38/sysdeps/unix/sysv/linux/powerpc/powerpc64/shlib-versions
+++ glibc-2.38-patched/sysdeps/unix/sysv/linux/powerpc/powerpc64/shlib-versions
@@ -1,7 +1,7 @@
 %ifdef HAVE_ELFV2_ABI
-DEFAULT			GLIBC_2.17
+DEFAULT			GLIBC_2.38
 ld=ld64.so.2
 %else
-DEFAULT			GLIBC_2.3
+DEFAULT			GLIBC_2.38
 ld=ld64.so.1
 %endif
--- glibc-2.38/sysdeps/unix/sysv/linux/riscv/shlib-versions
+++ glibc-2.38-patched/sysdeps/unix/sysv/linux/riscv/shlib-versions
@@ -1,14 +1,14 @@
 %if RISCV_ABI_XLEN == 64 && RISCV_ABI_FLEN == 64
-DEFAULT		GLIBC_2.27
+DEFAULT		GLIBC_2.38
 ld=ld-linux-riscv64-lp64d.so.1
 %elif RISCV_ABI_XLEN == 64 && RISCV_ABI_FLEN == 0
-DEFAULT		GLIBC_2.27
+DEFAULT		GLIBC_2.38
 ld=ld-linux-riscv64-lp64.so.1
 %elif RISCV_ABI_XLEN == 32 && RISCV_ABI_FLEN == 64
-DEFAULT		GLIBC_2.33
+DEFAULT		GLIBC_2.38
 ld=ld-linux-riscv32-ilp32d.so.1
 %elif RISCV_ABI_XLEN == 32 && RISCV_ABI_FLEN == 0
-DEFAULT		GLIBC_2.33
+DEFAULT		GLIBC_2.38
 ld=ld-linux-riscv32-ilp32.so.1
 %else
 %error cannot determine ABI
--- glibc-2.38/sysdeps/unix/sysv/linux/s390/s390-64/shlib-versions
+++ glibc-2.38-patched/sysdeps/unix/sysv/linux/s390/s390-64/shlib-versions
@@ -1,2 +1,2 @@
-DEFAULT			GLIBC_2.2
+DEFAULT			GLIBC_2.38
 ld=ld64.so.1
--- glibc-2.38/sysdeps/unix/sysv/linux/sh/shlib-versions
+++ glibc-2.38-patched/sysdeps/unix/sysv/linux/sh/shlib-versions
@@ -1,6 +1,6 @@
-libm=6			GLIBC_2.2
-libc=6			GLIBC_2.2
-ld=ld-linux.so.2	GLIBC_2.2
-libBrokenLocale=1	GLIBC_2.2
-libpthread=0		GLIBC_2.2
-libc_malloc_debug=0	GLIBC_2.2
+libm=6			GLIBC_2.38
+libc=6			GLIBC_2.38
+ld=ld-linux.so.2	GLIBC_2.38
+libBrokenLocale=1	GLIBC_2.38
+libpthread=0		GLIBC_2.38
+libc_malloc_debug=0	GLIBC_2.38
--- glibc-2.38/sysdeps/unix/sysv/linux/sparc/sparc64/shlib-versions
+++ glibc-2.38-patched/sysdeps/unix/sysv/linux/sparc/sparc64/shlib-versions
@@ -1,6 +1,6 @@
-libm=6			GLIBC_2.2
-libc=6			GLIBC_2.2
-ld=ld-linux.so.2	GLIBC_2.2
-libBrokenLocale=1	GLIBC_2.2
-libpthread=0		GLIBC_2.2
-libc_malloc_debug=0	GLIBC_2.2
+libm=6			GLIBC_2.38
+libc=6			GLIBC_2.38
+ld=ld-linux.so.2	GLIBC_2.38
+libBrokenLocale=1	GLIBC_2.38
+libpthread=0		GLIBC_2.38
+libc_malloc_debug=0	GLIBC_2.38
--- glibc-2.38/sysdeps/unix/sysv/linux/x86_64/64/shlib-versions
+++ glibc-2.38-patched/sysdeps/unix/sysv/linux/x86_64/64/shlib-versions
@@ -1,4 +1,4 @@
 # DEFAULT			Earliest symbol set
 # ---------------		------------------------------
-DEFAULT			GLIBC_2.2.5
+DEFAULT			GLIBC_2.38
 ld=ld-linux-x86-64.so.2
--- glibc-2.38/sysdeps/unix/sysv/linux/x86_64/x32/shlib-versions
+++ glibc-2.38-patched/sysdeps/unix/sysv/linux/x86_64/x32/shlib-versions
@@ -1,4 +1,4 @@
 # DEFAULT			Earliest symbol set
 # ---------------		------------------------------
-DEFAULT			GLIBC_2.16
+DEFAULT			GLIBC_2.38
 ld=ld-linux-x32.so.2
--- /dev/null
+++ glibc-2.38-patched/sysdeps/i386/fpu/k_standard.c
@@ -0,0 +1,9 @@
+/* Apparently needed for fmod.  */
+#include <math-svid-compat.h>
+
+#ifdef SHARED
+# undef LIBM_SVID_COMPAT
+# define LIBM_SVID_COMPAT 1
+#endif
+
+#include <sysdeps/ieee754/k_standard.c>
--- /dev/null
+++ glibc-2.38-patched/sysdeps/i386/fpu/k_standardf.c
@@ -0,0 +1,9 @@
+/* Apparently needed for fmod.  */
+#include <math-svid-compat.h>
+
+#ifdef SHARED
+# undef LIBM_SVID_COMPAT
+# define LIBM_SVID_COMPAT 1
+#endif
+
+#include <sysdeps/ieee754/k_standardf.c>
--- /dev/null
+++ glibc-2.38-patched/sysdeps/m68k/m680x0/fpu/k_standard.c
@@ -0,0 +1,9 @@
+/* Apparently needed for fmod.  */
+#include <math-svid-compat.h>
+
+#ifdef SHARED
+# undef LIBM_SVID_COMPAT
+# define LIBM_SVID_COMPAT 1
+#endif
+
+#include <sysdeps/ieee754/k_standard.c>
--- /dev/null
+++ glibc-2.38-patched/sysdeps/m68k/m680x0/fpu/k_standardf.c
@@ -0,0 +1,9 @@
+/* Apparently needed for fmod.  */
+#include <math-svid-compat.h>
+
+#ifdef SHARED
+# undef LIBM_SVID_COMPAT
+# define LIBM_SVID_COMPAT 1
+#endif
+
+#include <sysdeps/ieee754/k_standardf.c>
