--- a/Makefile.am
+++ b/Makefile.am
@@ -179,7 +179,7 @@
 	$(NM_V) $(NM) -u $< | sed 's/  *U  *//' | sort -u > $@
 MOSTLYCLEANFILES += gnumach-undef
 gnumach-undef-bad: gnumach-undef Makefile
-	$(AM_V_GEN) sed '$(foreach r,$(libgcc_routines) $(ld_magic_routines),/^$r$$/d;)' $< > $@
+	$(AM_V_GEN) sed '$(foreach r,$(libgcc_routines) $(ld_magic_routines) _GLOBAL_OFFSET_TABLE_,/^$r$$/d;)' $< > $@
 MOSTLYCLEANFILES += gnumach-undef-bad
 libgcc-routines.o: gnumach-undef gnumach-undef-bad
 	$(AM_V_at) if test -s gnumach-undef-bad; \
